packages/cli/src/__tests__/nps.test.ts

```diff
@@ -0,0 +1,194 @@
+import fs from 'fs'
+
+import { handleNpsSurveyImpl } from '../utils/nps/survey'
+
+const currentDate = new Date('2022-01-01T00:00:00.000Z')
+const laterDate = new Date('2023-01-01T00:00:00.000Z')
+const earlierDate = new Date('2021-01-01T00:00:00.000Z')
+const evenEarlierDate = new Date('2020-01-01T00:00:00.000Z')
+
+describe('nps survey', () => {
+  const originalEnv = { ...process.env }
+
+  let mockRead: jest.SpyInstance
+  let mockWrite: jest.SpyInstance
+
+  beforeAll(() => {
+    process.env = {}
+  })
+
+  afterAll(() => {
+    process.env = originalEnv
+  })
+
+  afterEach(() => {
+    mockRead?.mockRestore()
+    mockWrite?.mockRestore()
+  })
+
+  it('should exit immediately if running in CI', async () => {
+    mockRead = jest.spyOn(fs.promises, 'readFile').mockImplementation()
+    mockWrite = jest.spyOn(fs.promises, 'writeFile').mockImplementation()
+
+    const status = jest.fn()
+    const readline = {
+      question: jest.fn(),
+      write: jest.fn(),
+    }
+    const capture = jest.fn()
+
+    process.env.CI = 'true'
+    await handleNpsSurveyImpl(currentDate, { status }, readline, { capture })
+    delete process.env.CI
```

Comment:
I don't think we necessarily want to delete it, it might have had been set before us assigning `true` when the tests themselves are running on CI.
```suggestion
    process.env.CI = originalEnv.CI
```

Reply:
done

Reply:
note that the originalEnv is captured from before the test suite runs, and these individial tests are expected to run with `process.env = {}`, which is set from `beforeAll`/`beforeEach`, so I don't think it matters what we do here